src/space.cpp

```diff
@@ -495,13 +495,8 @@ static argval_t do_space(chunk_t *first, chunk_t *second, int &min_sp, bool comp
          break;
       }
 
-      // Issue #1005
```

Comment:
what happens with issue #1005 ?

Reply:
you say it's fixed and I don't see anything regressed. Ignore.

Reply:
friend void ::test::swap< >(future< T >&, future< T >&);
the space is removed between void and ::test because of option sp_before_dc = remove.
to fix the above issue code is written if  void previous chunk is friend add space after void.
which is causing issue #1288. latest changes will address the #1005 issue also which internally fixes #1288